backend/api/v1/routers/beneficiaries.py

```diff
@@ -64,50 +70,75 @@ async def import_beneficiary(
-                        beneficiary=b_id,
-                        action="Update",
-                        error=None,
-                    )
-                case other:
-                    logger.info(
-                        "block new beneficiary conflicting with existing beneficiaries: %s",
-                        [beneficiary.id for beneficiary in existing_rows],
-                    )
-                    return BeneficiaryImportResult(
-                        beneficiary=None,
-                        action="No action",
-                        error="Un bénéficiaire existant utilise cet internalId ou ce nom/prénom/date de naissance sur le territoire.",
-                    )
-
-        except Exception:
+            existing_rows: list[Beneficiary] = await get_beneficiaries_like(
+                db, beneficiary, deployment_id
+            )
+
+            if len(existing_rows) == 0:
+                beneficiary_id: UUID = await insert_beneficiary(
+                    db, beneficiary, deployment_id
+                )
+                new_notebook_id: UUID = await insert_notebook(
+                    db, beneficiary_id, beneficiary
+                )
+                await insert_wanted_jobs(db, new_notebook_id, beneficiary)
+                await insert_referent_and_structure(
+                    db, beneficiary_id, new_notebook_id, beneficiary
+                )
+                logger.info("inserted new beneficiary %s", beneficiary_id)
+                return BeneficiaryImportResult(
+                    beneficiary=beneficiary_id,
+                    action="Creation",
+                    error=None,
+                )
+
+            if (
+                len(existing_rows) == 1
+                and is_same_name(
+                    existing_rows[0].firstname,
+                    beneficiary.firstname,
+                    existing_rows[0].lastname,
+                    beneficiary.lastname,
+                )
+                and existing_rows[0].date_of_birth == beneficiary.date_of_birth
+                and existing_rows[0].internal_id == beneficiary.si_id
+                and existing_rows[0].deployment_id == deployment_id
+            ):
+                beneficiary_id = await update_beneficiary(
+                    db, beneficiary, deployment_id, existing_rows[0].id
+                )
+                notebook_id: UUID | None = await update_notebook(
+                    db, beneficiary_id, beneficiary
+                )
+                if notebook_id:
+                    await insert_wanted_jobs(db, notebook_id, beneficiary)
+                logger.info("updated existing beneficiary %s", beneficiary_id)
+                return BeneficiaryImportResult(
+                    beneficiary=beneficiary_id,
+                    action="Update",
+                    error=None,
+                )
+
+            logger.info(
+                "block new beneficiary conflicting with existing beneficiaries: %s",
```

Comment:
je comprend pas le message, on bloque un bénéficiaire ?

Reply:
@bChiquet tu as peut-être la réponse ?

Reply:
On empêche la création du bénéficiaire parce qu'il partage le trio nom/prénom/dob ou bien l'internal_id, mais pas l'ensemble.

Par exemple, un cas lundi, c'était une personne créée avec l'internal_id de quelqu'un d'autre qui existait déjà.

Reply:
je propose de remanier un peu le message d'erreur :
```suggestion
                "block beneficiary creation as it is conflicting with existing beneficiaries: %s",
```